### PR TITLE
Added staking logic to deposit stake. Next step will be to add withdrawing logic

### PR DESCRIPTION
**Pull request by leo-sizaret**

### CODE REVIEW

**Comment by muchiNoChi** [comment]
> **The latest updates on your projects**. Learn more about [Vercel for Git ↗︎](https://vercel.link/github-learn-more)
> 
> Name	Status	Preview	Comments	Updated (UTC)
> **starkpass**	✅ Ready ([Inspect](https://vercel.com/leo-sizaret/starkpass/9qbH2vTHBeavY7UR7Ttb5NqSMbMi))	[Visit Preview](https://vercel.live/open-feedback/starkpass-git-cairo-staking-leo-sizaret.vercel.app?via=pr-comment-visit-preview-link&passThrough=1)	💬 [**Add feedback**](https://vercel.live/open-feedback/starkpass-git-cairo-staking-leo-sizaret.vercel.app?via=pr-comment-feedback-link)	Jul 24, 2023 11:46pm
> **starkpass-dpgs**	✅ Ready ([Inspect](https://vercel.com/leo-sizaret/starkpass-dpgs/6gdxt3g69qiS9Hgjqwp8sCfe9mWS))	[Visit Preview](https://starkpass-dpgs-git-cairo-staking-leo-sizaret.vercel.app)		Jul 24, 2023 11:46pm

I'm not sure why we have 2 projects shown here. There's only one domain on Vercel project itself 🤔

**Comment by muchiNoChi** [comment]
> Visit Preview

Aaah, got it, it's for the PR branch deployment AND for the production (main) branch deployment.

File: src/event.cairo [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-trait IEventTrait<T> {
-    fn get_name(self: @T) -> felt252;
-    fn get_balance_of_contract(self: @T) -> u256;
-    fn get_attendant(self: @T, attendant: ContractAddress) -> bool;
-    fn get_ticket_price(self: @T) -> u256;
-    fn change_organizer(ref self: T, new_owner: ContractAddress);
-    fn transfer_balance_to_organizer(ref self: T);
-    fn buy_ticket(ref self: T);
-    fn send_tip(ref self: T, tip: u256);
+trait IEventTrait<TContractState> {
```
Comment (muchiNoChi) [inline src/event.cairo:5]: Is there a place to read about this type?

File: src/event.cairo [hunk 2]
```diff
@@ -64,6 +66,15 @@ mod StarkPassEvent {
         ticket_recipient: ContractAddress,
     }
 
+    // #### STRUCTS ####    
+    #[derive(Drop, storage_access::StorageAccess, Serde)]
```
Comment (muchiNoChi) [inline src/event.cairo:70]: What is a Serde?

**Review by muchiNoChi — APPROVED** [review]

File: src/event.cairo (hunk 1 above)
Comment (leo-sizaret) [reply src/event.cairo:5]: It's a generic trait, see here: https://www.cairo-lang.org/docs/language_constructs/generics.html

File: src/event.cairo (hunk 2 above)
Comment (leo-sizaret) [reply src/event.cairo:70]: Serde is a framework for serializing and deserializing data structures, see here: https://serde.rs/

**Pull request merged**